extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java

```diff
@@ -1194,37 +1211,44 @@ private PreGeneratedProxies generatedProxies(Set<String> entityClassNames, Index
         }
         try (ProxyBuildingHelper proxyHelper = new ProxyBuildingHelper(Thread.currentThread().getContextClassLoader())) {
             for (String entity : entityClassNames) {
-                Set<Class<?>> proxyInterfaces = new HashSet<>();
-                proxyInterfaces.add(HibernateProxy.class); //always added
-                Class<?> mappedClass = proxyHelper.uninitializedClass(entity);
-                String proxy = proxyAnnotations.get(entity);
-                if (proxy != null) {
-                    proxyInterfaces.add(proxyHelper.uninitializedClass(proxy));
-                } else if (!proxyHelper.isProxiable(mappedClass)) {
-                    //if there is no @Proxy we need to make sure the actual class is proxiable
-                    continue;
-                }
-                for (ClassInfo subclass : combinedIndex.getAllKnownSubclasses(DotName.createSimple(entity))) {
-                    String subclassName = subclass.name().toString();
-                    if (!entityClassNames.contains(subclassName)) {
-                        //not an entity
-                        continue;
-                    }
-                    proxy = proxyAnnotations.get(subclassName);
+                CachedProxy result;
+                if (proxyCache.cache.containsKey(entity) && !changedClasses.contains(entity)) {
+                    result = proxyCache.cache.get(entity);
```

Comment:
This looks very neat, but I believe you're not taking into account a change in related classes that would affect the output, such as `@Proxy(proxyClass -> changedInterface )` ?

Reply:
And what about the current class inheriting further from other changed classes?

Reply:
I now check the all classes and interfaces. If I check all interfaces then is that sufficient, or can @Proxy specify an interface the entity does not implement?